packages/next/export/index.ts

```diff
@@ -321,7 +321,12 @@ export default async function(
       Object.keys(prerenderManifest.routes).map(async route => {
         route = route === '/' ? '/index' : route
         const orig = join(distPagesDir, route)
-        const htmlDest = join(outDir, `${route}.html`)
+        const htmlDest = join(
+          outDir,
+          `${route}${
+            subFolders && route !== '/index' ? `${sep}index` : ''
```

Comment:
This should only activate for non `buildExport`, right? AFAIK this would do `index.html` during normal `next build`, which is undesirable (similar to how we ignore this setting for `next build` of auto static optimization pages).

Reply:
This is nested inside `if (!options.buildExport && prerenderManifest) {` so it's only triggered for non `next build`